### ISSUE TITLE
Flaky ZipFileExtractTest#test_extract_incorrect_size

### ISSUE THREAD

**Issue by voxik** [rubyzip/rubyzip#535]
The `ZipFileExtractTest#test_extract_incorrect_size` occasionally fails on [Fedora](https://koschei.fedoraproject.org/package/rubygem-rubyzip):

~~~
  1) Failure:
ZipFileExtractTest#test_extract_incorrect_size [/builddir/build/BUILD/rubyzip-2.3.2/usr/share/gems/gems/rubyzip-2.3.2/test/file_extract_test.rb:117]:
Expected false to be truthy.
~~~

**Comment by hainesr** [comment]
Does this happen frequently?
Does it happen with a clean checkout? I know some tests occasionally don't clear up as well as they might, so it might be an artefact of that.

I can't reproduce this on my Mac, and I don't think I've seen it myself under linux either.

**Comment by voxik** [comment]
> Does this happen frequently?

I'd say that this happens once per ~8 CI runs.

> Does it happen with a clean checkout?

The tests are executed above the installed gem and the environment is always fresh, so I'd say that the answer to your question is "yes"

> I know some tests occasionally don't clear up as well as they might, so it might be an artefact of that.
> 
> I can't reproduce this on my Mac, and I don't think I've seen it myself under linux either.

**Comment by voxik** [comment]
> if the order of test cases makes any difference ...

But it this was true, then it would be enough to use the same seed, rihgt? But that does not help to reproduce the issue

**Comment by voxik** [comment]
Actually, would it make any difference if the test suite failed on s390x, i.e. big endian, platform?

**Comment by voxik** [comment]
And this was fixed in #445 but not released yet.
